src/backend/optimizer.py

```diff
@@ -9,4 +14,28 @@ def __init__(self, producer):
 
     # The main function that optimizes the schedule. How the schedule and job should be implemented is up for discussion
     def optimize(self, schedule):
-        pass
+        objective = np.zeros(len(schedule))
+        return optimize.minimize(self.to_minimize, objective, tol=1.0, method="powell")
+
+    def to_minimize(self, schedule):
+        indices = list(set(chain.from_iterable(
+            map(lambda x: self.producer.schedule[x].load_profile.index.values.tolist(), range(0, len(schedule))))))
+        indices.sort()
```

Comment:
`indices` is a superset of all indices in the schedule. This is all the timestamps on which we need to check the consumption and production values for. It might be possible to do this in a smarter way.